### PR TITLE
Updated Arch Bishop skills from kRO maintenance

### PR DESCRIPTION
**Pull request by aleos89**
* **Addressed Issue(s)**: #3231

* **Server Mode**: Pre-renewal and Renewal

* **Description of Pull Request**: 
  * Updated Arch Bishop skills from kRO maintenance: http://ro.gnjoy.com/news/update/View.asp?seq=222&curpage=1.
  * Added two new skills: Vituperatum and Convenio.
Thanks to @secretdataz and @Rytech2 for the cast time info!

### CODE REVIEW

File: db/re/skill_tree.txt [hunk 1]
```diff
@@ -2612,6 +2612,8 @@
 4057,2515,5,2044,1,2053,1,0,0,0,0,0,0 //AB_SECRAMENT#Secrament#
 4057,5011,5,2051,2,0,0,0,0,0,0,0,0 //AB_OFFERTORIUM#Offertorium#
 4057,5014,5,0,0,0,0,0,0,0,0,0,0 //ALL_FULL_THROTTLE#Full Throttle#
+4057,5072,5,2044,1,2053,1,0,0,0,0,0,0 //AB_VITUPERATUM#Vituperatum#
+4057,5073,1,2031,1,2046,5,0,0,0,0,0,0 //AB_CONVENIO#Convenio#
```
Comment (Angelic234) [inline db/re/skill_tree.txt]: 2031 should be 2039

Comment (aleos89) [reply db/re/skill_tree.txt]: You're right, but it should be 2046, not 2039.

**Comment by ecdarreola** [comment]
Having problem with the skill vit uperatum.
image

**Comment by ecdarreola** [comment]
Having problem with the skill vit uperatum.
image

File: db/re/skill_require_db.txt [hunk 2]
```diff
@@ -1054,7 +1054,7 @@
 
 // New Arch Bishop Skills
 5072,0,0,144:120:106:92:78,0,0,0,99,0,0,none,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0				//AB_VITUPERATUM##
-5073,0,0,70,0,0,0,99,0,0,none,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0								//AB_CONVENIO##
+5073,0,0,70,0,0,0,99,0,0,none,0,0,717,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0								//AB_CONVENIO##
```
Comment (Angelic234) [inline db/re/skill_require_db.txt]: Skill 5073 doesn't need item 717.
Skill 5072 needed.
info : https://www.divine-pride.net/database/skill/5072

Comment (Angelic234) [reply db/re/skill_require_db.txt]: I tested skill AB_CONVENIO always warp partymember to wrong map

**Comment by Angelic234** [comment]
Skill AB_VITUPERATUM animation should be ===> https://i.imgur.com/luZ0SCT.gif

**Comment by laziem** [comment]
Got this with Convenio
convenio

I think Lauda Ramus effect is Critical Damage instead of Critical.

Expiato the mdef ignore part seems not working from my test. Target still receive same magic damage with or without Expiatio.

Did Vituperatum and Convenio need newer client? Mine is 2017-05-17a. I already add Vituperatum & Convenio at skilldescript.lub, skillid.lub, skillinfolist.lub, skilltreeview.lub, skilltreeview_web.lub, but the skill shows up at 2nd Job skill tree instead of 3rd Job.

And the skill somewhat after learn it, turn gray but consume skill point. So far I can only use the skill only from @useskill, but no animation shows https://i.imgur.com/luZ0SCT.gif for Vituperatum.
Is it something wrong at my end?

File: db/re/item_db.txt [hunk 3]
```diff
@@ -6553,7 +6553,7 @@
 12330,Recall_FemaleGM,Summon Female GameMaster Scroll,2,0,,10,,,,,0xFFFFFFFF,63,2,,,,,,{ mercenary_create 2001,1800000; },{},{}
 12331,Ginseng,Ginseng,0,0,,10,,,,,0xFFFFFFFF,63,2,,,,,,{ percentheal 6,0; },{},{}
 12332,Fruit_Juice,Fruit Juice,0,0,,10,,,,,0xFFFFFFFF,63,2,,,,,,{ percentheal 0,6; },{},{}
-12333,Ansila,Ancilla,2,0,,10,,,,,0xFFFFFFFF,63,2,,,,,,{ percentheal 0,15; },{},{}
+12333,Ansila,Ancilla,2,0,,10,,,,,0xFFFFFFFF,63,2,,,,,,{ percentheal 0,15; autobonus "{ bonus bHealPower,30; bSPrecovRate,15; }",100,60000; },{},{}
```
Comment (xTakami) [inline db/re/item_db.txt]: Power heal 30% is not working.

File: src/map/skill.cpp [hunk 4]
```diff
@@ -10941,7 +10941,7 @@ int skill_castend_nodamage_id (struct block_list *src, struct block_list *bl, ui
 
 	case AB_VITUPERATUM:
 		if (flag&1)
-			sc_start(src, bl, type, 100, skill_lv, skill_get_time(skill_id, skill_lv));
+			clif_skill_nodamage(bl, bl, skill_id, skill_lv, sc_start(src, bl, type, 100, skill_lv, skill_get_time(skill_id, skill_lv)));
```
Comment (redlightliu) [inline src/map/skill.cpp]: It should be
clif_skill_nodamage("src", bl, skill_id, skill_lv, sc_start(src, bl, type, 100, skill_lv, skill_get_time(skill_id, skill_lv)));
not
clif_skill_nodamage("bl", bl, skill_id, skill_lv, sc_start(src, bl, type, 100, skill_lv, skill_get_time(skill_id, skill_lv)));

File: src/map/status.cpp [hunk 5]
```diff
@@ -6192,6 +6181,8 @@ static signed short status_calc_critical(struct block_list *bl, struct status_ch
 		critical -= sc->data[SC__UNLUCKY]->val2;
 	if(sc->data[SC_BEYONDOFWARCRY])
 		critical += sc->data[SC_BEYONDOFWARCRY]->val3;
+	if(sc->data[SC_LAUDARAMUS])
+		critical += 5 * sc->data[SC_LAUDARAMUS]->val1;
```
Comment (redlightliu) [inline src/map/status.cpp]: This effect should be increase "Critical Damage"
Not increase "Critical"

**Comment by Lemongrass3110** [comment]
@spinzaku this is a data side problem. It works without problems with 2018 kRO client and data.

**Comment by spinzaku** [comment]
So it works with 2018 EXE ? but not 2017? hmmm my data files are latest

**Comment by Lemongrass3110** [comment]
2017-06-28:
grafik

2018-06-20:
grafik

So definitely a clientside problem.

**Comment by xTakami** [comment]
It seems that you are not ignoring the magical defense, could someone confirm if it is working well?

			if (tsc && tsc->data[SC_EXPIATIO]) {
				short i = 5 * tsc->data[SC_EXPIATIO]->val1; // 5% per level

				i = min(i, 100); //cap it to 100 for 0 mdef min
				mdef = (mdef * (100 - i)) / 100;
				//mdef2 = (mdef2 * (100 - i)) / 100;
			}

Thanks!

**Comment by xTakami** [comment]
After the last change when using ancilla, you get a state of permanent stone curse.

**Comment by laziem** [comment]
Lauda Ramus, when wear Holy armor and got status Howling of Mandragora, only the caster (AB) that have effect removed, the other party member (wear holy armor) still got the Howling of Mandragora

**Comment by aleos89** [comment]
@laziem Is official behavior to stop this skill then or what? Guess we'd need to test on kRO.

**Comment by laziem** [comment]
@aleos89 Sorry for confusing. I mean, it should remove Howling of Mandragora status to all party member too, even with Holy armor, because the caster Howling of Mandragora status is removed too when use Holy armor.

**Comment by aleos89** [comment]
@laziem I confirmed it on kRO that it does indeed continue on through even with holy armor. The reason this happens is because of this check:
```cpp
	if (src!=bl && type > -1 &&
		(i = skill_get_ele(skill_id, skill_lv)) > ELE_NEUTRAL &&
		skill_get_inf(skill_id) != INF_SUPPORT_SKILL &&
		battle_attr_fix(NULL, NULL, 100, i, tstatus->def_ele, tstatus->ele_lv) <= 0)
		return 1; //Skills that cause an status should be blocked if the target element blocks its element.
```
It's unrelated to this PR though as I didn't adjust the element to Lauda Ramus, so it'd be best to create an issue so it's separate from this PR.

**Comment by admkakaroto** [comment]
```
Program terminated with signal 11, Segmentation fault.
#0  0x000000000058c227 in skill_calc_heal (src=src@entry=0x890da50, target=target@entry=0x7deb4f0,
    skill_id=skill_id@entry=28, skill_lv=skill_lv@entry=10, heal=heal@entry=true) at skill.cpp:552
552                                     hp_bonus += sc->data[SC_ANCILLA]->val1;
Missing separate debuginfos, use: debuginfo-install glibc-2.17-106.el7_2.8.x86_64 libgcc-4.8.5-11.el7.x86_64 libstdc++-4.8.5-11.el7.x86_64 pcre-8.32-15.el7_2.1.x86_64 zlib-1.2.7-17.el7.x86_64
(gdb) bt full
#0  0x000000000058c227 in skill_calc_heal (src=src@entry=0x890da50, target=target@entry=0x7deb4f0,
    skill_id=skill_id@entry=28, skill_lv=skill_lv@entry=10, heal=heal@entry=true) at skill.cpp:552
        skill = <optimized out>
        hp = 1950
        hp_bonus = 34
        tsd = 0x7deb4f0
        sc = 0x890df50
        global_bonus = 1
        sd = 0x890da50
        tsc = 0x7deb9f0
#1  0x00000000005b45f4 in skill_castend_nodamage_id (src=0x890da50, bl=0x7deb4f0,
    skill_id=<optimized out>, skill_lv=10, tick=53194992, flag=0) at skill.cpp:6167
        heal = <optimized out>
        heal_get_jobexp = <optimized out>
        dstsd = <optimized out>
        dstmd = 0x0
        hd = <optimized out>
        sstatus = 0x890dee4
        tsc = 0x7deb9f0
        tsce = 0x0
        sd = <optimized out>
        mer = 0x0
        tstatus = 0x7deb984
        type = <optimized out>
        __FUNCTION__ = "skill_castend_nodamage_id"
        md = <optimized out>
        i = <optimized out>
#2  0x00000000005d03a5 in skill_castend_id (tid=<optimized out>, tick=tick@entry=53194992,
    id=<optimized out>, data=data@entry=0) at skill.cpp:11368
        fail = false
        res = -1 '\377'
        sc = <optimized out>
        flag = 0
        target = <optimized out>
```


Crash...

**Comment by admkakaroto** [comment]
I love you aleos-san.

**Comment by xTakami** [comment]
Thanks @aleos89 <3!

**Comment by Pizzaguy15** [comment]
don't know if this post belongs here but when I try to compile I get this error on skill.cpp line 10970:
compileerror

**Comment by aleos89** [comment]
You merged to master incorrectly.

**Comment by systemD2540** [comment]
Animation vituperatum not have effect after cast

**Comment by zackdreaver** [comment]
@aleos89 
Will be there any problem on server side if I changed the skill ID of new skills both client side and server db?
Is there any "src" modification needed?

**Comment by aleos89** [comment]
The constant in skill.hpp needs to match that new ID then.

**Comment by zackdreaver** [comment]
Ok, thanks, I'll try it.

**Comment by mjonrest** [comment]
screenvicit-ro world000

cant find those 2 new skill.. someone know how to fix it?

**Pull request merged**